packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -197,9 +197,10 @@ async function fetchLatestSnapshotCore(
         assert(storageToken !== null, 0x1e5 /* "Storage token should not be null" */);
 
         let controller: AbortController | undefined;
+        let fetchTimeout: ReturnType<typeof setTimeout> | undefined;
         if (snapshotOptions?.timeout !== undefined) {
             controller = new AbortController();
-            setTimeout(
+            fetchTimeout = setTimeout(
                 () => controller!.abort(),
                 snapshotOptions.timeout,
             );
```

Comment:
I'd move this block down closer to PerformanceEvent.timedExecAsync() call, to reduce amount of code in between (any exception in between will result in timer hanging around and not being cancelled)